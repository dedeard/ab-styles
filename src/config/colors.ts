export const colors = {
  gray_100: '#f7fafc',
  gray_200: '#edf2f7',
  gray_300: '#e2e8f0',
  gray_400: '#cbd5e0',
  gray_500: '#a0aec0',
  gray_600: '#718096',
  gray_700: '#4a5568',
  gray_800: '#2d3748',
  gray_900: '#1a202c',
  red_100: '#fff5f5',
  red_200: '#fed7d7',
  red_300: '#feb2b2',
  red_400: '#fc8181',
  red_500: '#f56565',
  red_600: '#e53e3e',
  red_700: '#c53030',
  red_800: '#9b2c2c',
  red_900: '#742a2a',
  indigo_100: '#ebf4ff',
  indigo_200: '#c3dafe',
  indigo_300: '#a3bffa',
  indigo_400: '#7f9cf5',
  indigo_500: '#667eea',
  indigo_600: '#5a67d8',
  indigo_700: '#4c51bf',
  indigo_800: '#434190',
  indigo_900: '#3c366b',
};

export default colors;
