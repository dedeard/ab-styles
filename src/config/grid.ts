export const grid = {
  space: 1,
  container: {
    xs: 'full',
    sm: 'full',
    md: 45,
    lg: 60,
    xl: 71.25,
  },
  columns: {
    xs: 4,
    sm: 8,
    md: 12,
    lg: 12,
    xl: 12,
  },
  breakpoints: {
    xs: 1,
    sm: 33.75,
    md: 48,
    lg: 64,
    xl: 71.25,
  },
};

export default grid;

export type Grid = typeof grid;
