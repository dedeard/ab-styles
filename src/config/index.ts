export { BREAKPOINTS, DEFAULT_CONFIG } from './constants';
export { default as config } from './config';
