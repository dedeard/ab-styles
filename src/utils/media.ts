import { config } from '../config';

import { Breakpoints } from '../types/emotion';

export function media(breakpoint: Breakpoints): string {
  const breakpoints = config().grid.breakpoints;

  if (!Object.keys(breakpoints).includes(breakpoint)) {
    throw new Error(`Breakpoint '${breakpoint}' not found`);
  }

  return `@media (min-width: ${breakpoints[breakpoint]}em)`;
}
