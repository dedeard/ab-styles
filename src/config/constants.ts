import { DefaultTheme, Breakpoints } from '../types/emotion';
import colors from './colors';
import grid from './grid';

export const COLORS = colors;

export const GRID = grid;

export const BREAKPOINTS: Array<Breakpoints> = ['xs', 'sm', 'md', 'lg', 'xl'];

export const DEFAULT_CONFIG: DefaultTheme = {
  grid,
  colors,
};
