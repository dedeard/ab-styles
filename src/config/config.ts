import { Theme } from '@emotion/react';
import { DEFAULT_CONFIG } from './constants';
import { DefaultTheme } from '../types/emotion';

type ThemeArgs = DefaultTheme | { theme: DefaultTheme };

function isTheme(args: ThemeArgs): args is Theme {
  return (args as { theme: Theme }).theme === undefined;
}

export const getTheme = (args: ThemeArgs): Theme =>
  isTheme(args) ? args : args.theme;

function config(props: ThemeArgs = DEFAULT_CONFIG): DefaultTheme {
  const theme = getTheme(props);

  return {
    ...DEFAULT_CONFIG,
    ...theme,
  };
}

export default config;
