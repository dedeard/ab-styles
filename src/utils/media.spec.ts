import { responsive } from './media';

import { config } from '../config';

describe('responsive', () => {
  it('should return the for each breakpoint', () => {
    const result = responsive(
      config()?.grid?.breakpoints,
      () => `
        background: red;
      `
    );

    expect(result).toMatchSnapshot();
  });
});
