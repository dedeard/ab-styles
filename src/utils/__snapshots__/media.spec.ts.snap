// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responsive should return the for each breakpoint 1`] = `
Array [
  Object {
    "map": undefined,
    "name": "2er7l4",
    "next": undefined,
    "styles": "
      @media (min-width: 1em) {
        
        background: red;
      
      }
    ",
    "toString": [Function],
  },
  Object {
    "map": undefined,
    "name": "1ace4ch",
    "next": undefined,
    "styles": "
      @media (min-width: 33.75em) {
        
        background: red;
      
      }
    ",
    "toString": [Function],
  },
  Object {
    "map": undefined,
    "name": "q7k0m7",
    "next": undefined,
    "styles": "
      @media (min-width: 48em) {
        
        background: red;
      
      }
    ",
    "toString": [Function],
  },
  Object {
    "map": undefined,
    "name": "199et9d",
    "next": undefined,
    "styles": "
      @media (min-width: 64em) {
        
        background: red;
      
      }
    ",
    "toString": [Function],
  },
  Object {
    "map": undefined,
    "name": "1gzbfso",
    "next": undefined,
    "styles": "
      @media (min-width: 71.25em) {
        
        background: red;
      
      }
    ",
    "toString": [Function],
  },
]
`;
